### PR TITLE
feat: add WithSpanFromContext logrus helper

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: logrusx/helper.go [hunk 1]
```diff
@@ -107,6 +107,25 @@ func (l *Logger) WithRequest(r *http.Request) *Logger {
 	return ll
 }
 
+func (l *Logger) WithSpan(sc trace.Span) *Logger {
```
Comment (zepatrik) [inline logrusx/helper.go:110]: Did you consider passing the full context here instead and then using `trace.SpanFromContext`? I'd guess that that would be the most universal usage.

File: logrusx/helper.go [hunk 2]
```diff
@@ -107,6 +107,25 @@ func (l *Logger) WithRequest(r *http.Request) *Logger {
...
+	if sc == nil {
+		return l
+	}
+	spanCtx := sc.SpanContext()
+	if !spanCtx.IsValid() {
+		return l
+	}
+
+	traces := map[string]string{}
```
Comment (zepatrik) [inline logrusx/helper.go:119]:
```suggestion
	traces := make(map[string]string, 2)
```
as we know the max capacity already.

File: logrusx/helper.go (hunk 1 above)
Comment (jonas-jonas) [reply logrusx/helper.go]: Done.

File: logrusx/helper.go (hunk 2 above)
Comment (jonas-jonas) [reply logrusx/helper.go]: Done, it was copied from the request helper..

**Pull request merged**